myserver.cpp

```diff
@@ -0,0 +1,553 @@
+    }
+    else if(msg.at(0)=="ChangeCategoryName"){
+            ChangeCategoryName(msg);
+    }
+    else if(msg.at(0)=="AddNote"){
+            AddNote(msg);
+    }
+    else if(msg.at(0)=="GetNote"){
+            GetNote(msg);
+    }
+    else if(msg.at(0)=="ChangeCategoryId"){
+            ChangeCategoryId(msg);
+    }
+    else if(msg.at(0)=="SetHeader"){
+            SetHeader(msg);
+    }
+    else if(msg.at(0)=="SetText"){
+            SetText(msg);
+    }
+    else if(msg.at(0)=="AddFriendAccess"){
+            AddFriendAccess(msg);
+    }
+    else if(msg.at(0)=="CheckFriendAccess"){
+            CheckFriendAccess(msg);
+    }
+    else if(msg.at(0)=="DeleteFriendAccess"){
+            DeleteFriendAccess(msg);
+    }
+    else{
+            socketWrite("Unclear command");
+    }
+}
+void CMyServer::AddUser(const std::vector <std::string> &msg){
+    int iRtCode = IDataBase::AddUser(msg.at(1), msg.at(2), iUserId);
+    QJsonObject AddUser;
+    switch(iRtCode){
+    case SUCCESS:
+        AddUser.insert("RequestType", "AddUser");
+        AddUser.insert("SUCCESS", iUserId);
+        document.setObject(AddUser);
+        bytes = document.toJson( QJsonDocument::Indented );
+        socketWrite(bytes);
+        break;
+    case INVALID_INPUT_DATA:
+        socketWrite("INVALID_INPUT_DATA");
+        break;
+    case UNEXPECTED_ERROR:
+        socketWrite("UNEXPECTED_ERROR");
+        break;
+    }
+}
+void CMyServer::GetUserId(const std::vector <std::string> &msg){
+    int iRtCode = IDataBase::GetUserId(msg.at(1), msg.at(2), iUserId);
+    QJsonObject GetUserId;
+    switch(iRtCode){
+    case SUCCESS:
+        GetUserId.insert("RequestType", "GetUserId");
+        GetUserId.insert("SUCCESS", iUserId);
+        document.setObject(GetUserId);
+        bytes = document.toJson( QJsonDocument::Indented );
+        socketWrite(bytes);
+        break;
+    case NO_SUCH_USER:
+        socketWrite("NO_SUCH_USER");
+        break;
+    case UNEXPECTED_ERROR:
+        socketWrite("UNEXPECTED_ERROR");
+        break;
+    }
+}
+void CMyServer::GetFriendList(const std::vector <std::string> &msg){
+    std::vector <std::string> FriendList;
+    int iRtCode = IDataBase::GetFriendList(iId, FriendList);
+    QJsonObject GetFriendList;
+    std::string sList;
+
+    switch(iRtCode){
+    case SUCCESS:
+        for(int i = 0; i <= FriendList.size(); i++){
+             sList += FriendList[i] + '|';
+        }
+        GetFriendList.insert("RequestType", "GetFriendList");
+        GetFriendList.insert("SUCCESS", sList.data());
+        document.setObject(GetFriendList);
+        bytes = document.toJson( QJsonDocument::Indented );
+        socketWrite(bytes);
+        break;
+    case NO_SUCH_USER:
+        socketWrite("NO_SUCH_USER");
+        break;
+    case UNEXPECTED_ERROR:
+        socketWrite("UNEXPECTED_ERROR");
+        break;
+    }
+}
+void CMyServer::AddFriend(const std::vector <std::string> &msg){
+    iUserId = std::stoi(msg.at(1));
```

Comment:
Same here, there aren't checks for valid data, so it can cause exceptions

Reply:
Fixed